app/models/katello/alternate_content_source.rb

```diff
@@ -35,7 +35,9 @@ class AlternateContentSource < Katello::Model
     validates :base_url, :subpaths, :upstream_username,
               :upstream_password, if: :simplified?, absence: true
     validates :base_url, if: -> { custom? || rhui? }, presence: true
-    validates :products, if: -> { custom? || rhui? }, absence: true
+    validates :products, if: -> { custom? || rhui? }, absence: {
+      message: "should not be set for any custom or rhui ACS"
```

Comment:
Cannot seems better for validations IMO..
```suggestion
      message: "cannot be set for any custom or rhui ACS"
```